### PR TITLE
Uncaught FileNotFoundException when deleting bts file

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
This problem is very mysterious. The logs show that the file has been streamed to a peer and then deleted. And then 4 seconds later node tries to stream and delete the same file again. The only explanation that comes to my mind it that the file has been enqueued twice to the stream queue.

For now, I did:
1. Improved the code so it anticipates the case when the file is missing and don’t throws an exception.
2. Added a debug message each time a packet file gets enqueued. Next time we can check if a packet got enqueued twice for streaming.

Caveat:
Since there will be no exception thrown anymore we need to monitor the logs for stream errors.

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3443

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### CODE REVIEW

File: comm/src/main/scala/coop/rchain/comm/transport/PacketOps.scala [hunk 1]
```diff
@@ -1,52 +1,58 @@
...
-                   gracefullyClose(fin) >> Left(unableToRestorePacket(file, th)).pure[F]
-                 case Right(packet) => gracefullyClose(fin) >> Right(packet).pure[F]
-               }
-    } yield resErr
+  def restore[F[_]: Sync](file: Path): F[CommErr[Packet]] = {
+    def fileInputStream: F[CommErr[FileInputStream]] =
+      Sync[F]
+        .delay(new FileInputStream(file.toFile))
+        .attempt
+        .map(_.fold(unableToRestorePacket(file, _).asLeft, _.asRight))
```
Comment (itegulov) [inline comm/src/main/scala/coop/rchain/comm/transport/PacketOps.scala]: This should be a `Resource`

File: comm/src/main/scala/coop/rchain/comm/transport/PacketOps.scala [hunk 2]
```diff
@@ -1,52 +1,58 @@
...
-                   case Left(th) =>
-                     gracefullyClose(fos) >> Left(unableToStorePacket(packet, th)).pure[F]
-                   case Right(_) =>
-                     gracefullyClose(fos) map {
-                       case Left(th) => Left(unableToStorePacket(packet, th))
-                       case Right(_) => Right(file)
-                     }
-                 }
-      } yield resErr
+        closeErr <- gracefullyClose(fos)
```
Comment (itegulov) [inline comm/src/main/scala/coop/rchain/comm/transport/PacketOps.scala]: I would say `createPacketFile` should return `Resource[F, PacketFile]` as well but it probably falls out of scope of this particular PR

Comment (sebtomba) [reply comm/src/main/scala/coop/rchain/comm/transport/PacketOps.scala]: `StreamHandler` is using this function too. So maybe next time.

**Pull request merged**